Lib/unittest/mock.py

```diff
@@ -1378,6 +1378,10 @@ def decorate_async_callable(self, func):
 
     def get_original(self):
         target = self.getter()
+        if type(target) != type:
```

Comment:
This is a code path for `mock.patch` . `patch.object` passes a custom getter function which is just an identity function to return the target. The check should be done there. This would also need a test where a string passed to `patch.object` to assert `TypeError` being raised.

I can't think of a usecase where someone wants to pass a string and patch it's method. I would wait for others since the doc is clear that an object is needed and if we need to be more strict.

Reply:
That's the code that is raising an error in my case. Why would it raise the error otherwise?

Reply:
I did! I passed a string instead of a class, because that's how you would usually patch a function or something else. You pass the string. Like `patch('module.function')`.

Reply:
I think that what @tirkarthi meant is that a user might want to pass an instance of str and patch a method of `str`. But that should probably not be a concern, as even if unlikely, the methods of `str` area all readonly:
```
>>> patch.object("str_instance", "split").start()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/bb/lib/python3.7/unittest/mock.py", line 1450, in start
    result = self.__enter__()
  File "/opt/bb/lib/python3.7/unittest/mock.py", line 1410, in __enter__
    setattr(self.target, self.attribute, new_attr)
AttributeError: 'str' object attribute 'split' is read-only
```

Reply:
I didn't know know string's attributes are already read-only. Thanks Mario for clarification.